File: mercor-scd-client-library/pkg/models/payment/payment_line_item.go
// pkg/models/payment/payment_line_item.go
package payment

import (
	"time"

	"github.com/mercor-ai/scd-go-client/pkg/models"
)

// PaymentLineItem represents a payment line item entity
type PaymentLineItem struct {
	models.BaseSCDEntity
	JobUID     string  `gorm:"column:job_uid;not null"`
	TimelogUID string  `gorm:"column:timelog_uid;not null"`
	Amount     float64 `gorm:"column:amount;not null;type:decimal(10,2)"`
	Status     string  `gorm:"column:status;not null"`
}

// TableName returns the table name for the entity
func (p *PaymentLineItem) TableName() string {
	return "payment_line_items"
}

// CloneForNewVersion creates a new version of the payment line item
func (p *PaymentLineItem) CloneForNewVersion(uid string, version int, now time.Time) models.SCDEntity {
	return &PaymentLineItem{
		BaseSCDEntity: models.BaseSCDEntity{
			ID:        p.ID,
			Version:   version,
			UID:       uid,
			CreatedAt: now,
			UpdatedAt: now,
		},
		JobUID:     p.JobUID,
		TimelogUID: p.TimelogUID,
		Amount:     p.Amount,
		Status:     p.Status,
	}
}

File: src/main/java/com/mercor/assignment/scd/domain/paymentlineitem/model/PaymentLineItem.java
package com.mercor.assignment.scd.domain.paymentlineitem.model;

import com.mercor.assignment.scd.domain.core.model.AbstractSCDEntity;
import com.mercor.assignment.scd.domain.core.model.SCDEntity;
import com.mercor.assignment.scd.domain.job.model.Job;
import com.mercor.assignment.scd.domain.timelog.model.Timelog;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.ForeignKey;
import jakarta.persistence.Index;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;
import java.math.BigDecimal;
import java.util.Date;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

/**
 * Entity class for PaymentLineItem
 */
@Data
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder
@Entity
@Table(name = "payment_line_items")
public class PaymentLineItem extends AbstractSCDEntity {

  /**
   * The UID of the job associated with the payment line item
   */
  @Column(name = "job_uid", nullable = false)
  private String jobUid;

  /**
   * The UID of the timelog associated with the payment line item
   */
  @Column(name = "timelog_uid", nullable = false)
  private String timelogUid;

  /**
   * The amount of the payment
   */
  @Column(name = "amount", nullable = false, precision = 10, scale = 2)
  private BigDecimal amount;

  /**
   * The status of the payment (e.g., "paid", "not-paid")
   */
  @Column(name = "status", nullable = false)
  private String status;

  /**
   * The job entity associated with the payment line item This is the JPA relationship to the Job entity
   */
  @ManyToOne
  @JoinColumn(
      name = "job_uid",
      referencedColumnName = "uid",
      insertable = false,
      updatable = false
  )
  private Job job;

  /**
   * The timelog entity associated with the payment line item This is the JPA relationship to the Timelog entity
   */
  @ManyToOne
  @JoinColumn(
      name = "timelog_uid",
      referencedColumnName = "uid",
      insertable = false,
      updatable = false
  )
  private Timelog timelog;

  @Override
  public PaymentLineItem cloneForNewVersion(String uid, int version, Date now) {
    return PaymentLineItem.builder()
        .id(this.getId())
        .version(version)
        .uid(uid)
        .createdAt(now)
        .updatedAt(now)
        .timelogUid(this.getTimelogUid())
        .jobUid(this.getJobUid())
        .amount(this.getAmount())
        .status(this.getStatus())
        .build();
  }
}